core/state/iterator_test.go

```diff
@@ -51,12 +51,15 @@ func TestNodeIteratorCoverage(t *testing.T) {
 			t.Errorf("state entry not reported %x", hash)
 		}
 	}
-	for _, key := range db.TrieDB().DiskDB().(*ethdb.MemDatabase).Keys() {
+	it := db.TrieDB().DiskDB().(ethdb.Database).NewIterator()
```

Comment:
I hate to add a last minute suggestion, but if this is a common pattern, it may be worth considering an `Database.forEach(func(key, value []byte)error) error` method that encapsulates the iteration and release:

```
func (db *Database) forEach(f func(key, value []byte)error) error {
  it := db.NewIterator()
  defer it.Release()
  for it.Next() {
    if err := f(it.Key(), it.Value()); err != nil {
      return err
    }
  }
  return nil
}
```

Reply:
I think I'd rather avoid this for now at least. I'm more of a fan of explicit code and one way of doing things. Lets see if this appears a few more times in the database rewrites and extensions, and if it makes sense we can always add it then.